include/eve/arch/has_native_support.hpp

```diff
@@ -0,0 +1,20 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/spec.hpp>
+#include <type_traits>
+
+namespace eve
+{
+  //! @brief Indicates if a given callable is natively supported for type T on a given API
+  template<typename Callable, typename T, typename API = current_api_type>
+  struct has_native_support : std::true_type
+  {};
```

Comment:
Ooofff.. I think this is an overly generic thing that we won't be able to support properly.